share/tools/web_config/webconfig.py

```diff
@@ -1315,17 +1314,19 @@ def do_POST(self):
             return self.send_error(403)
         self.path = p
 
-        ctype, pdict = cgi.parse_header(self.headers["content-type"])
+        # This is cheesy, we want just the actual content-type.
+        # In some cases it'll give us the encoding as well,
+        # ("application/json;charset=utf-8")
+        # but we don't currently care.
+        ctype = self.headers["content-type"].split(";")[0]
 
-        if ctype == "multipart/form-data":
-            postvars = cgi.parse_multipart(self.rfile, pdict)
-        elif ctype == "application/x-www-form-urlencoded":
+        if ctype == "application/x-www-form-urlencoded":
             length = int(self.headers["content-length"])
             url_str = self.rfile.read(length).decode("utf-8")
             postvars = parse_qs(url_str, keep_blank_values=1)
         elif ctype == "application/json":
             length = int(self.headers["content-length"])
-            url_str = self.rfile.read(length).decode(pdict["charset"])
+            url_str = self.rfile.read(length).decode("utf-8")
```

Comment:
I can't think of any security implications this has (if you're authenticated to the server, you can just send fish code to run anyway) but it might be worth making a note to that effect.

Reply:
What I'd expect to happen here is that, if anything shows up that isn't valid utf-8, this fails to decode it, which triggers an exception (a "[UnicodeError](https://docs.python.org/3/library/exceptions.html#UnicodeError)"), which at the very least causes webconfig to exit out of this method (since it's not caught), if not entirely.

So if anyone ever stumbles on this, I'd expect a bug report. And tbh I don't even expect that (I'm not actually sure if we handle anything except for utf-8 and ascii - which is a utf-8 subset - correctly to begin with).

As for malicious input, you're right - you could just send well-formed input and have it be executed.

What kind of note would you like?

Reply:
Just something to that effect - that we're aware of the assumption that anything sent is Unicode, that this may not be true in the wild but to the best of our knowledge it is, and that there shouldn't be any security implications because of the checks performed earlier in the function.